bin/message_factory/message_factory.py

```diff
@@ -28,51 +28,65 @@
 
 
 class MessageFactory:
-    """This class serves to automatically send an email to a group of admin/developers.
-    """
+    """This class serves to automatically email a group of admin/developers."""
 
-    def __init__(self, config_path=os.environ['YANGCATALOG_CONFIG_PATH']):
+    def __init__(
+            self,
+            config_path=os.environ['YANGCATALOG_CONFIG_PATH'],
+            close_connection_after_message_sending: bool = True,
+    ):
         config = create_config(config_path)
-        self.__email_from = config.get('Message-Section', 'email-from')
-        self.__is_production = config.get('General-Section', 'is-prod')
-        self.__is_production = True if self.__is_production == 'True' else False
-        self.__email_to = config.get('Message-Section', 'email-to').split()
-        self.__developers_email = config.get('Message-Section', 'developers-email').split()
+        self._email_from = config.get('Message-Section', 'email-from')
+        self._is_production = config.get('General-Section', 'is-prod') == 'True'
+        self._email_to = config.get('Message-Section', 'email-to').split()
+        self._developers_email = config.get('Message-Section', 'developers-email').split()
         self._temp_dir = config.get('Directory-Section', 'temp')
-        self.__me = config.get('Web-Section', 'domain-prefix')
-        self.__me = self.__me.split('/')[-1]
-        self.__smtp = smtplib.SMTP('localhost')
+        self._me = config.get('Web-Section', 'domain-prefix').split('/')[-1]
+        self._smtp = smtplib.SMTP('localhost')
+        self.close_connection_after_message_sending = close_connection_after_message_sending
```

Comment:
This should also be prefixed with an underscore. If this got manually modified after the initialization, the smtp connection could stay open after the MessageFactory is destructed.